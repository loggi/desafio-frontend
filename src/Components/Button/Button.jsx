// import { StyledButton } from "./Styled";

// const Button = ({bgColor, textColor, children}) => {
//     return <StyledButton bgColor={bgColor} textColor={textColor}>{children}</StyledButton>;
//   };
  
//   export default Button;