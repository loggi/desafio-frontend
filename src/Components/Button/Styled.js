// import styled from "styled-components";

// export const StyledButton = styled.button.attrs({
//     textColor: "#fff",
//     bgColor: "#006AF0"
// })`
//   border: none;
//   border-radius: 10px;
//   padding: 10px 0;
//   width: 200px;
//   font-weight: 700;
//   color: ${({textColor}) => textColor};
//   background-color: ${({bgColor}) => bgColor};
// `;