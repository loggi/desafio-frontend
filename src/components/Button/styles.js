import styled from '@emotion/styled'
import { css } from '@emotion/react'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    align-items: center;
    background-color: ${theme.palette.primary.contrastText};
    border: none;
    border-radius: 100px;
    box-shadow: rgba(72, 79, 95, 0.06) 0px 2.25px 10.5px;
    color: ${theme.palette.primary.main};
    cursor: pointer;
    display: flex;
    flex-direction: row;
    font-family: ${theme.typography.fontFamily};
    font-size: ${theme.typography.button.fontSize};
    height: 48px;
    justify-content: center;
    margin: 0 auto;
    max-width: 100%;
    overflow: hidden;
    padding: 14px 30px;
    position: relative;
    text-align: center;
    transition: all 0.2s ease 0s;
    width: 300px;

    &:hover {
      background-color: ${theme.palette.primary.light};
      border: none;
      color: ${theme.palette.primary.contrastText};
    }

    &:active {
      box-shadow:
        rgba(190, 199, 214, 0.02) 0px 0.345043px 0.690086px -0.172521px,
        rgba(159, 171, 187, 0.03) 0px 0.690086px 0.862607px,
        rgba(72, 79, 95, 0.06) 0px 0.172521px 1.72521px;
    }

    span {
      margin-right: 10px;
    }

    svg {
      width: 15px;
      height: 15px;
    }
  `}
`
