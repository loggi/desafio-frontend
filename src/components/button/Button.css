.button {
    padding: 16px;
    line-height: 1.6;
    border: 1px solid var(--color-common-light);
    border-radius: 4px;
    text-transform: uppercase;
    font-size: 14px;
    font-weight: 600;
    background-color: transparent;
    color: var(--color-common-light);
    cursor: pointer;
    transition: background-color 0.5s ease;
}

.button:hover {
    border-color: var(--color-common-light);
    background-color: var(--color-opacity-ligth);
    background-color: var(--color-opacity-ligth);
}

.button:disabled {
    border-color: var(--color-opacity-ligth);
    background-color: var(--color-opacity-ligth);
    color: var(--color-opacity-ligth);
    cursor: not-allowed;
}

.button--small {
    padding: 8px 14px
}

.button--large {
    padding: 24px;
    font-size: 16px;
    font-weight: 700;
}

@media (max-width: 700px) {
    .button--large {
        width: 100%;
    }
}
