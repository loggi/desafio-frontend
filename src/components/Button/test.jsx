import { screen } from '@testing-library/react'

import { render } from '@/utils/tests/helpers'

import Button from '.'

describe('<Button />', () => {
  it('should render the Button', () => {
    const { container } = render(<Button>Button</Button>)

    expect(screen.getByRole('button', { name: /Button/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
