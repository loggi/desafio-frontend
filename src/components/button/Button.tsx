import "./Button.css"

interface ButtonProps {
  children: React.ReactNode
  disabled?: boolean
  onClick: React.ReactEventHandler
  size?: string
}

export const Button = ({
  children,
  disabled,
  onClick,
  size = "",
  ...rest
}: ButtonProps) => {
  return (
    <button
      className={size ? `button button--${size}` : `button`}
      onClick={onClick}
      disabled={disabled}
      {...rest}
    >
      {children}
    </button>
  )
}
