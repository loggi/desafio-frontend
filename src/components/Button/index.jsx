import PropTypes from 'prop-types'
import { ChevronRightIcon } from '@heroicons/react/24/outline'

import * as S from './styles'

const Button = ({ children, ...props }) => (
  <S.Wrapper as="button" {...props}>
    <span>{children}</span>
    <ChevronRightIcon />
  </S.Wrapper>
)

Button.propTypes = {
  children: PropTypes.string.isRequired
}

export default Button
