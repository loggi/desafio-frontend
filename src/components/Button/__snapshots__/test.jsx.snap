// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render the heading 1`] = `
.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #FFFFFF;
  border: none;
  border-radius: 100px;
  box-shadow: rgba(72, 79, 95, 0.06) 0px 2.25px 10.5px;
  color: #006AFF;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  font-family: fontFamily;
  font-size: 16;
  height: 48px;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  margin: 0 auto;
  max-width: 100%;
  overflow: hidden;
  padding: 14px 30px;
  position: relative;
  text-align: center;
  -webkit-transition: all 0.2s ease 0s;
  transition: all 0.2s ease 0s;
  width: 300px;
}

.emotion-0:hover {
  background-color: #0095FF;
  border: none;
  color: #FFFFFF;
}

.emotion-0:active {
  box-shadow: rgba(190, 199, 214, 0.02) 0px 0.345043px 0.690086px -0.172521px,rgba(159, 171, 187, 0.03) 0px 0.690086px 0.862607px,rgba(72, 79, 95, 0.06) 0px 0.172521px 1.72521px;
}

.emotion-0 span {
  margin-right: 10px;
}

.emotion-0 svg {
  width: 15px;
  height: 15px;
}

<button
  class="emotion-0"
>
  <span>
    Button
  </span>
  <svg
    aria-hidden="true"
    fill="none"
    stroke="currentColor"
    stroke-width="1.5"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M8.25 4.5l7.5 7.5-7.5 7.5"
      stroke-linecap="round"
      stroke-linejoin="round"
    />
  </svg>
</button>
`;
